### PR TITLE
Feat/add claim for

### PR DESCRIPTION
**Pull request by razww**
Changes include:

* remove old logic
* add `claimWithdrawFor` function to allow to claim withdrawals for other users
* add role to pause the contract

### CODE REVIEW

File: contracts/ListaStakeManager.sol [hunk 1]
```diff
@@ -194,17 +230,15 @@ contract ListaStakeManager is
...
         require(bnbToWithdraw > minBnb, "Bnb amount is too small to withdraw");
 
-        uint256 totalAmount = bnbToWithdraw;
-        uint256 totalAmountInSlisBnb = _amountInSlisBnb;
-        if (withdrawalQueue.length != 0) {
-            totalAmount += withdrawalQueue[requestIndexMap[requestUUID]].totalAmount;
-            totalAmountInSlisBnb += withdrawalQueue[requestIndexMap[requestUUID]].totalAmountInSlisBnb;
-        }
+        // TODO: use withdrawalQueue[length-1]
+        require(withdrawalQueue.length != 0, "Queue is empty");
```
Comment (razww) [inline contracts/ListaStakeManager.sol:239]: use if condition instead of require

File: contracts/ListaStakeManager.sol [hunk 2]
```diff
@@ -236,92 +270,48 @@ contract ListaStakeManager is
...
-        // 2. queue.length > 0 && uuid < queue[0].uuid => old logic
-        // 3. queue.length > 0 && uuid >= queue[0].uuid => new logic
-
-        if (withdrawalQueue.length != 0 && uuid >= withdrawalQueue[0].uuid) {
-            // new logic
-            UserRequest storage request = withdrawalQueue[requestIndexMap[uuid]];
-            require(uuid < nextConfirmedRequestUUID, "Not able to claim yet");
-            amount = request.amount;
-        } else {
-            // old logic
```
Comment (razww) [inline contracts/ListaStakeManager.sol:261]: keep old logic here, ensure that old requests are still claimable

File: contracts/ListaStakeManager.sol [hunk 3]
```diff
@@ -367,28 +356,8 @@ contract ListaStakeManager is
...
-            botRequest.endTime = block.timestamp;
-            undelegatedQuota -= botRequest.amount;
-            coveredAmount += botRequest.amount;
-            coveredSlisBnbAmount += botRequest.amountInSnBnb;
-            ++nextConfirmedRequestUUID;
-        }
 
-        // new logic, new requests exist; `withdrawalQueue[0].uuid <= nextConfirmedRequestUUID` condition can be removed in next version
-        if (withdrawalQueue.length != 0 && withdrawalQueue[withdrawalQueue.length - 1].uuid >= nextConfirmedRequestUUID && withdrawalQueue[0].uuid <= nextConfirmedRequestUUID) {
+        if (withdrawalQueue.length != 0 && withdrawalQueue[0].uuid <= nextConfirmedRequestUUID) {
```
Comment (razww) [inline contracts/ListaStakeManager.sol:360]: double check if should use 
`withdrawalQueue.length != 0 && withdrawalQueue[withdrawalQueue.length - 1].uuid >= nextConfirmedRequestUUID`

File: contracts/ListaStakeManager.sol [hunk 4]
```diff
@@ -25,27 +26,45 @@ contract ListaStakeManager is
 {
     using SafeERC20Upgradeable for IERC20Upgradeable;
 
-    uint256 public totalSnBnbToBurn; // received User withdraw requests; no use in new logic
+    // Deprecated variable; always zero
+    uint256 public totalSnBnbToBurn;
```
Comment (razww) [inline contracts/ListaStakeManager.sol:30]: rename to `placeholder`

**Review by qingyang-lista — APPROVED** [review]
LGTM

**Pull request merged**